enhancements/dns/dns-operator-operand-logging-level.md

```diff
@@ -0,0 +1,239 @@
+So,
+
+`logLevel: "Normal"`  will enable the "errors" class: `log . { class error }`.
+
+`logLevel: "Debug"` will enable the "denial" class: `log . { class denial error }`.
+
+`logLevel: "Trace"` will enable the "all" class: `log . { class all }`.
+
+Note that the `errors` plugin is always enabled.  The `errors` plugin logs TCP/UDP connection errors whereas `log . { class error }` logs DNS error responses (such as SERVFAIL).
+The CoreDNS reloads its configuration without requiring a restart, so the operator can adjust CoreDNS's log level just by updating the Corefile configmap without need to restart the pod.
+
+
+We will be adding an API field `operatorlogLevel` in DNSSpec for the type DNSLogLevel
+```go
+// operatorLogLevel controls the logging level of the DNS Operator.
+// Valid values are: "Normal", "Debug", "Trace".
+// Defaults to "Normal".
+// setting operatorLogLevel: Trace will produce extremely verbose logs.
+// +optional
+// +kubebuilder:default=Normal
+OperatorLogLevel DNSLogLevel `json:"operatorLogLevel,omitempty"`
+```
+This new field would allow a cluster administrator to specify the desired logging level specifically for the DNS Operator.
+
+Additionally, a new `LogLevel` of type `DNSLogLevel` will be added for CoredDNS logging :
+```go
+// logLevel describes the desired logging verbosity for CoreDNS.
+// Any one of the following values may be specified:
+// * Normal logs errors from upstream resolvers.
+// * Debug logs errors, NXDOMAIN responses, and NODATA responses.
+// * Trace logs errors and all responses.
+//  Setting logLevel: Trace will produce extremely verbose logs.
+// Valid values are: "Normal", "Debug", "Trace".
+// Defaults to "Normal".
+// +optional
+// +kubebuilder:default=Normal
+LogLevel DNSLogLevel `json:"logLevel,omitempty"`
+```
+
+Both of these new APIs would use the exisiting `DNSLogLevel` definitions:
+
+```go
+
+// +kubebuilder:validation:Enum:=Normal;Debug;Trace
+type DNSLogLevel string
+
+var (
+// Normal is the default.  Normal, working log information, everything is fine, but helpful notices for auditing or common operations.  In kube, this is probably glog=2.
+DNSLogLevelNormal DNSLogLevel = "Normal"
+
+// Debug is used when something went wrong.  Even common operations may be logged, and less helpful but more quantity of notices.  In kube, this is probably glog=4.
+DNSLogLevelDebug DNSLogLevel = "Debug"
+
+// Trace is used when something went really badly and even more verbose logs are needed.  Logging every function call as part of a common operation, to tracing execution of a query.  In kube, this is probably glog=6.
+DNSLogLevelTrace DNSLogLevel = "Trace"
+)
+
+```
+
+### User Stories
+
+* As an OpenShift Cluster Administrator, I want to be able to raise the logging level of the DNS Operator and CoreDNS so that I can more quickly
+track down OpenShift DNS issues.
+
+* Some users recently added the new alert 'CoreDNS is returning SERVFAIL for X% of requests alert' to the recent updates of OCP.
```

Comment:
What do you mean by "Some users" added the alert?  Is this referring to an alert that is in the product?

Reply:
no some customers add prometheus alerts so they will need this logging to be set